autogpt_platform/frontend/src/components/integrations/credentials-input.tsx

```diff
@@ -148,14 +151,69 @@ export const CredentialsInput: FC<{
     }
   }, [credentials, selectedCredentials, onSelectCredentials]);
 
-  const singleCredential = useMemo(() => {
-    if (!credentials || !("savedCredentials" in credentials)) return null;
+  // Helper function to extract host from URL
+  const getHostFromUrl = (url: string): string | null => {
+    try {
+      const urlObj = new URL(url);
+      return urlObj.hostname;
+    } catch {
+      return null;
+    }
+  };
 
-    if (credentials.savedCredentials.length === 1)
-      return credentials.savedCredentials[0];
+  // Get current host from siblingInputs
+  const currentUrl = siblingInputs?.url;
```

Comment:
The `CredentialsField` has a `discriminator` parameter for exactly this purpose; this relation with the `url` input shouldn't be hard-coded here.